File: snakecode/0001.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        want = {}
        for i, v in enumerate(nums):
            if v not in want:
                want[target - v] = i
            else:
                return [want[v], i]
        return []

if __name__ == '__main__':
    s = Solution()
    nums = [3, 2, 5, 9, -4, 0, 7]
    target = 1
    print(s.twoSum(nums, target))

File: ktcode/src/main/kotlin/h01/0001.kt
package h01

class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        for ((i, v) in nums.withIndex()) {
            map[v]?.let { return intArrayOf(it, i)}
            map[target-v] = i
        }
        return intArrayOf()
    }
}